quick fix: a blank plugin option should be true, not false

niftynei opened the pull request:
Fix a syntactic mistake i made in #3582 

Also fix the option interop test so that it actually verifies the log messages.

rustyrussell commented:
Yes, as @darosior notes, pretty sure this doesn't work, since --foo will not get a callback with opt_register_arg.

We'd almost need a new type to mean a 'simple' flag argument, say `simple`?  It would use opt_register_noarg instead.

We deliberately don't allow optional arguments, since they're ambigious (is `--foo --bar` to be interpreted as `--foo=--bar` or two options?).

niftynei commented:
why does this test pass then?

https://github.com/ElementsProject/lightning/blob/f66f1a8f0c0be942b67777e5d87be64ad3f5faa3/tests/test_plugin.py#L66-L74

niftynei commented:
> Didn't test yet, but we stil use opt_register_arg() to register plugin options. With an empty string passed is it equivalent to opt_unregister_noarg() ?

Hm so the tests seem to pass, but I'm not sure that the command line arguments will work the same as the python mediated options? I removed it entirely, I think it's fine to force specification.

> We'd almost need a new type to mean a simple flag argument, say flag? It would use opt_register_noarg instead.

This seemed like a fun thing to add, so I went ahead and added this as a type.
